### PR TITLE
Dumforms code review1

### PR DESCRIPTION
**Pull request by dumforms**
Program Analysis:
This program is using RoomDB to store slice files. The intended purpose is to track the time and amount of printing material used by a particular print, although this functionality has not been coded yet. The program was available on time, and compiled successfully.

The documentation was inconsistent. Some files like edu.uc.kovaciad.slicetracker.dto.Material contained excellent documentation that even made use of comment tags. Many other files lacked such thorough documentation.

### CODE REVIEW

**Comment by dumforms** [comment]
Links to 3 commits in my project:
https://github.com/erswagner7/PuzzlePic/commit/cd732d9e30b59cedcf28bcd79cfd063b62cf0654
https://github.com/erswagner7/PuzzlePic/commit/9bfe0175c546ff95ee43330120ae2f7e4d6fb48c
https://github.com/erswagner7/PuzzlePic/commit/b7ad865bd3d58d2a9f9daac678eb6fed3f10c812

3 technical concepts learned:
1. Kotlin class layout
2. var retrofit in class RetrofitInstance should be declared null (I thought this was unnecessary, reviewed class videos, learned that they did it correctly)
3. comment tags (like @param) are discouraged but not forbidden

File: app/src/main/java/edu/uc/kovaciad/slicetracker/ui/main/MainFragment.kt [hunk 1]
```diff
@@ -12,16 +12,13 @@ import edu.uc.kovaciad.slicetracker.R
 
 class MainFragment : Fragment() {
 
-    companion object {
-        fun newInstance() = MainFragment()
-    }
-
```
Comment (dumforms) [inline app/src/main/java/edu/uc/kovaciad/slicetracker/ui/main/MainFragment.kt:18]: Moved companion object to bottom of class to comply with Kotlin best practices: https://kotlinlang.org/docs/coding-conventions.html#class-layout

File: app/src/main/java/edu/uc/kovaciad/slicetracker/ui/main/MainFragment.kt [hunk 2]
```diff
@@ -12,16 +12,13 @@ import edu.uc.kovaciad.slicetracker.R
...
     private lateinit var viewModel: MainViewModel
     // For whatever reason, it won't compile when I do lateinit for this, but will if I don't
 //    private lateinit var applicationViewModel: ApplicationViewModel
 
     override fun onCreateView(
-        inflater: LayoutInflater, container: ViewGroup?,
+        inflater: LayoutInflater,
+        container: ViewGroup?,
```
Comment (dumforms) [inline app/src/main/java/edu/uc/kovaciad/slicetracker/ui/main/MainFragment.kt:21]: Expanded function signature to comply with Kotlin best practices: https://kotlinlang.org/docs/coding-conventions.html#functions

File: app/src/test/java/edu/uc/kovaciad/slicetracker/DataUnitTest.kt [hunk 3]
```diff
@@ -26,7 +26,7 @@ class DataUnitTest {
 //    }
 
     @Test
-    fun twoplustwo() {
+    fun twoPlusTwo() {
```
Comment (dumforms) [inline app/src/test/java/edu/uc/kovaciad/slicetracker/DataUnitTest.kt:29]: This one is very minor, but I renamed the function to comply with Kotlin naming standards: https://kotlinlang.org/docs/coding-conventions.html#function-names

File: app/src/test/java/edu/uc/kovaciad/slicetracker/ExampleUnitTest.kt [hunk 4]
```diff
@@ -1,17 +0,0 @@
...
- * Example local unit test, which will execute on the development machine (host).
- *
- * See [testing documentation](http://d.android.com/tools/testing).
- */
-class ExampleUnitTest {
-    @Test
-    fun addition_isCorrect() {
-        assertEquals(4, 2 + 2)
-    }
-}
```
Comment (dumforms) [inline app/src/test/java/edu/uc/kovaciad/slicetracker/ExampleUnitTest.kt:17]: This class had become redundant. The addition_isCorrect() test was recreated in the DataUnitTest class as twoPlusTwo(). With no other content present, it made sense to remove the class altogether and write all relevant tests in the DataUnitTest class.

**Comment by discospiff** [comment]
The changes here can either be merged as a branch, or implemented individually, or not used at all.  I'll leave it up to the group to decide.

**Pull request closed without merging**